modules/nw-coredns-egress-firewall.adoc

```diff
@@ -0,0 +1,44 @@
+// Module included in the following assemblies:
+//
+// * networking/openshift_network_security/configuring-egress-firewall-ovn.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="nw-coredns-egress-firewall_{context}"]
+= Improved DNS resolution and resolving wildcard domain names
+
+There might be situations where the IP addresses associated with a DNS record change frequently or you might want to specify wildcard domain names in your egress firewall policy rules.
+
+:FeatureName: Improved DNS resolution for egress firewall rules
+include::snippets/technology-preview.adoc[]
+
+In this situation, the OVN-K cluster manager creates a `DNSNameResolver` custom resource object for each unique DNS name used in your egress firewall policy rules. This custom resource stores the following information:
+
+[source,yaml]
+.Example `DNSNameResolver` custom resource definition
+----
+apiVersion: networking.openshift.io/v1alpha1
+kind: DNSNameResolver
+spec:
+  name: www.example.com. <1>
+status:
+  resolvedNames:
+  - dnsName: www.example.com. <2>
+    resolvedAddress:
+    - ip: "1.2.3.4" <3>
+      ttlSeconds: 60 <4>
+      lastLookupTime: "2023-08-08T15:07:04Z" <5>
+----
+<1> the DNS name, this can either be a standard DNS name or a wildcard DNS name. For a wildcard DNS name, the DNS name resolution information contains all the DNS names that match the wildcard DNS name.
+<2> the resolved DNS name matching the `spec.name` field <1>. If the `spec.name` field contains a wildcard DNS name, then multiple `dnsName` entries are created that contain the standard DNS names that match the wildcard DNS name when resolved. If the wildcard DNS name can also be successfully resolved, then this field will also store the wildcard DNS name.
+<3> the current IP addresses associated with the DNS name
+<4> the last time-to-live (TTL) duration
+<5> the last lookup time
+
+If during DNS resolution the DNS name in the query matches any name defined in a `DNSNameResolver` CR, then the above information is updated accordingly in the CR `status` field. For unsuccessful DNS wildcard name lookups, the request is retried after a default TTL of 30 minutes.
+
+The OVN-K cluster manager watches for updates to an `EgressFirewall` custom resource object, and will either create, modify or delete `DNSNameResolver` custom resources associated with those egress firewall policies when one such update occurs.
+
+[WARNING]
+====
+It is not recommended to modify `DNSNameResolver` custom resources directly. Doing so might lead to undesired behavior of your egress firewall.
```

Comment:
```suggestion
Do not modify `DNSNameResolver` custom resources directly, because this can lead to unwanted behavior of your egress firewall.
```
[SSG](https://redhat-documentation.github.io/supplementary-style-guide/#recommend) suggests avoiding "recommends" language. [IBM SG](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#d) says do not use "desired" (and I guess, undesired)